### ISSUE TITLE
NPE in ActivityTestRule.launchActivity

### ISSUE THREAD

**Issue by gkylafas** [robolectric/robolectric#4649]
### Description
Running my tests from Android Studio with robolectric 4.2 & 4.1 works fine.
However, when run from the command line (`./gradlew :app:testDebugUnitTest`) with robolectric 4.2,  some fail with the following stacktrace (same for all cases):

```java
> java.lang.NullPointerException
	at org.robolectric.android.fakes.RoboMonitoringInstrumentation.startActivitySync(RoboMonitoringInstrumentation.java:44)
	at androidx.test.rule.ActivityTestRule.launchActivity(ActivityTestRule.java:358)
	at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:529)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:256)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Running the same tests with robolectric 4.1 from command line is 100% successful.
The results are the same whether `android.enableUnitTestBinaryResources` is `true` or `false`.

### Steps to Reproduce
Run `./gradlew :app:testDebugUnitTest`

### Robolectric & Android Version
robolectric 4.2
android gradle plugin 3.3.1

### Link to a public git repo demonstrating the problem:
N/A

**Comment by xian** [comment]
Same issue, different stacktrace, as #4640.

**Comment by gkylafas** [comment]
@xian, I am not sure this is the same issue. I have already defined a test Activity
```kotlin
class EmptyTestActivity : AppCompatActivity()
```
 *and* added it to my `app/src/test/AndroidManifest.xml`
```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest
    xmlns:android="http://schemas.android.com/apk/res/android"
    package="my.package.test">

    <application>
        <activity
            android:name=".EmptyTestActivity"
            android:theme="@style/AppTheme" />
    </application>
</manifest>
```

I then defined a base class for my fragment tests
```kotlin
@RunWith(AndroidJUnit4::class)
abstract class FragmentUnitTests {
    @get:Rule
    val rule = ActivityTestRule(EmptyTestActivity::class.java)
}
```
and every test class extending from `FragmentUnitTests` fails in 4.2, while succeeding in 4.1, from the command line.
Am I doing something wrong?

**Comment by romkal** [comment]
I'd say to make sure, that this manifest is actually used in the test.

The difference between 4.1 and 4.2 is, that 4.1 would return `ActivityInfo` even for activities that doesn't exist in the manifest, while 4.2 will actually consult the manifest.

Maybe this is something in your gradle build file?

It would be great if you could provide a git repo with actual sample.

**Comment by shoheikawano** [comment]
Hi, I have made a sample repo for reproducing this NPE. 
It would be great if I misunderstand something or misplace the manifest.
Thanks in advance!

https://github.com/shoheikawano/Robolectric4.2Sample/tree/master/app/src/test/java/com/shoheikawano/robolectricsample

**Comment by gkylafas** [comment]
Actually, the sample repo of @shoheikawano reproduces #4640.
However, if you apply the following patch to it, it reproduces this issue as well.

```diff
diff --git i/app/build.gradle w/app/build.gradle
index ee2e418..83ac9c3 100644
--- i/app/build.gradle
+++ w/app/build.gradle
@@ -33,6 +33,7 @@ dependencies {
     testImplementation 'org.robolectric:robolectric:4.2'
     testImplementation 'androidx.test:core:1.1.0'
     testImplementation 'androidx.test.ext:junit:1.1.0'
+    testImplementation("androidx.test:rules:1.1.1")
     androidTestImplementation 'androidx.test:runner:1.1.1'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
 }
diff --git i/app/src/test/java/com/shoheikawano/robolectricsample/ExampleUnitTest.kt w/app/src/test/java/com/shoheikawano/robolectricsample/ExampleUnitTest.kt
index 214f5a6..c4768e0 100644
--- i/app/src/test/java/com/shoheikawano/robolectricsample/ExampleUnitTest.kt
+++ w/app/src/test/java/com/shoheikawano/robolectricsample/ExampleUnitTest.kt
@@ -1,13 +1,16 @@
 package com.shoheikawano.robolectricsample
 
-import androidx.test.core.app.ActivityScenario
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
 import org.junit.Assert.assertEquals
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class ExampleUnitTest {
+    @get:Rule
+    val rule = ActivityTestRule(TestActivity::class.java)
 
     /*
         Running test with following command: ./gradlew app:testDebugUnitTest
@@ -34,8 +37,6 @@ class ExampleUnitTest {
      */
     @Test
     fun addition_isCorrect() {
-        ActivityScenario.launch(TestActivity::class.java).onActivity {
-            assertEquals(4, 2 + 2)
-        }
+        assertEquals(4, 2 + 2)
     }
 }
```

**Comment by shoheikawano** [comment]
As I mentioned here: https://github.com/robolectric/robolectric/issues/4640#issuecomment-468939893, I understand and agree with the idea that robolectric follows its behavior as application running on the device, but I also would like to know if there is a way to put all the activity and test related class inside test package, which requires loading AndroidManifest.xml inside test package when testing.

**Comment by romkal** [comment]
OK. So it actually seems that it is working as expected.

The `TestActivity` you are trying to use is not in the manifest. If you turn binary resources on, then you will pick the manifest at `src/main/AndroidManifest.xml` but there is nothing that would pick up the manifest at `src/test/AndroidManifest.xml`.

In fact there is no way right now to specify an extra manifest for the robolectric tests in gradle (though in bazel you can use `android_local_test.manifest` to pass an extra manifest to Robolectric tests.

**Comment by gkylafas** [comment]
OK, moving my `EmptyTestActivity` under the `src/main` source set and including it in `src/main/AndroidManifest.xml` fixes this.

**Comment by xian** [comment]
Workaround documented here: #4736
Gradle fix tracked here: #4726.
